### DISCUSSION TITLE
Wait for a page to be loaded after `cy.click()`

### DISCUSSION THREAD

**Discussion by hamidmayeli** [cypress-io/cypress#27243] [Questions and Help] [answered]
I have the following code:

```ts
Cypress.Commands.add("login", (username: string, password: string) => {
    cy.get("a").contains("Login").click(); // this line takes the user to a new page

    cy.get("label").contains("Username").next("input").type(username);

    cy.get("label").contains("Password").next("input").type(password);
    
    cy.get("button").contains("Login").click();
  });
```

After simulating the first click, user should be taken to a new page, however, the next line happens before loading.

I have tested a solution that I could find (calling `cy.visit()`), but it results in an error.
`must be called with a url or an options object containing a url as its 1st argument`

Is there a way to wait for the new page? Something like:
```ts
cy.waitToSee(cy.get("label").contains("Username"));
```

**Comment by adnanerlansyah403** [comment, marked as answer]
you can use the ones event of cypress, like this :

cy.on('url:changed', (url) => {
 // will get the new currentUrl from callback while the url was changed
})
